tests/test_model_builder.py

```diff
@@ -15,15 +15,19 @@
 class TestModelBuilder:
     @pytest.fixture
     async def pgpool(self, postgresql):
-        pool = await asyncpg.create_pool(
-            host=postgresql.info.host,
-            database=postgresql.info.dbname,
-            user=postgresql.info.user,
-            password=postgresql.info.password,
-            port=postgresql.info.port,
+        conn_string = (
+            f"host={postgresql.info.host} "
+            f"dbname={postgresql.info.dbname} "
+            f"user={postgresql.info.user} "
+            f'password="{postgresql.info.password}" '
+            f"port={postgresql.info.port}"
         )
-        assert pool
-        yield pool
+
+        pool = psycopg_pool.AsyncConnectionPool(
+            conn_string, min_size=2, max_size=10, open=False
+        )
+        await pool.open()
+        await pool.wait()
         await pool.close()
```

Comment:
It seems that this fixture was useless here, since you're not returning the pool anymore (the yield is missing), but the tests still work.

Reply:
Gotta leave there else I would need to create it in each function when passing it down to the model builder, it wont create an error per se right now, but we need to pass `psycopg_pool.AsyncConnectionPool` type object.
I tried it still and pyright also doesn't let me so will leave it there.

Reply:
But the thing is that `pgpool` returns `None` here. I double checked in debug to make sure.

Reply:
Weird I did tried sending None directly although it didnt seem to fail, pyright wasnt liking it at all